File: lib/lang/hu.js


function join_with_shared_prefix(a, b, joiner) {
  let m = a;
  let i = 0;

  // HACK: This gets around "today through on Tuesday" or cases like it, which
  // are incorrect in English.
  if(m === "ma" || m === "holnap")
    m = "a" + m + "i nap";

  while(i !== m.length &&
        i !== b.length &&
        m.charCodeAt(i) === b.charCodeAt(i))
    ++i;

  while(i && m.charCodeAt(i - 1) !== 32)
    --i;

  return a + " és " + b.slice(i) + joiner;
}

module.exports = {
  "clear": "derült",
  "no-precipitation": "csapadékmentes idő",
  "mixed-precipitation": "vegyes csapadék",
  "possible-very-light-precipitation": "szitáló csapadék lehetséges",
  "very-light-precipitation": "szitáló csapadék",
  "possible-light-precipitation": "gyenge csapadék lehetséges",
  "light-precipitation": "gyenge csapadék",
  "medium-precipitation": "csapadék",
  "heavy-precipitation": "intenzív csapadék",
  "possible-very-light-rain": "gyenge szitálás lehetséges",
  "very-light-rain": "gyenge szitálás",
  "possible-light-rain": "gyenge eső lehetséges",
  "light-rain": "gyenge eső",
  "medium-rain": "eső",
  "heavy-rain": "intenzív eső",
  "possible-very-light-sleet": "ónos szitálás lehetséges",
  "very-light-sleet": "ónos szitálás",
  "possible-light-sleet": "gyenge ónos eső lehetséges",
  "light-sleet": "gyenge ónos eső",
  "medium-sleet": "ónos eső",
  "heavy-sleet": "havas eső",
  "possible-very-light-snow": "hószállingózás lehetséges",
  "very-light-snow": "hószállingózás",
  "possible-light-snow": "gyenge havazás lehetséges",
  "light-snow": "gyenge havazás",
  "medium-snow": "havazás",
  "heavy-snow": "intenzív havazás",
  "possible-thunderstorm": "zivatarok lehetséges",
  "thunderstorm": "zivatarok",
  "possible-medium-precipitation": "csapadék lehetséges",
  "possible-heavy-precipitation": "intenzív csapadék lehetséges",
  "possible-medium-rain": "eső lehetséges",
  "possible-heavy-rain": "intenzív eső lehetséges",
  "possible-medium-sleet": "ónos eső lehetséges",
  "possible-heavy-sleet": "havas ónos eső lehetséges",
  "possible-medium-snow": "havazás lehetséges",
  "possible-heavy-snow": "intenzív havazás lehetséges",
  "possible-freezing-drizzle": "fagyos gyenge szitálás lehetséges",
  "freezing-drizzle": "fagyos gyenge szitálás",
  "possible-light-freezing-rain": "gyenge fagyos eső lehetséges",
  "light-freezing-rain": "gyenge fagyos eső",
  "possible-freezing-rain": "fagyos eső lehetséges",
  "freezing-rain": "fagyos eső",
  "possible-heavy-freezing-rain": "intenzív fagyos eső lehetséges",
  "heavy-freezing-rain": "intenzív fagyos eső",
  "possible-hail": "fagyos lehetséges",
  "hail": "fagyos",
  "light-wind": "enyhe szél",
  "medium-wind": "mérsékelt szél",
  "heavy-wind": "erős szél",
  "low-humidity": "száraz idő",
  "high-humidity": "párás idő",
  "fog": "ködös idő",
  "very-light-clouds": "erős derült",
  "light-clouds": "közepes felhősödés",
  "medium-clouds": "erős felhősödés",
  "heavy-clouds": "borult idő",
  "today-morning": "ma reggel",
  "later-today-morning": "ma késő délelőtt",
  "today-afternoon": "ma délután",
  "later-today-afternoon": "ma késő délután",
  "today-evening": "ma este",
  "later-today-evening": "ma késő este",
  "today-night": "ma éjjel",
  "later-today-night": "ma késő éjjel",
  "tomorrow-morning": "holnap reggel",
  "tomorrow-afternoon": "holnap délután",
  "tomorrow-evening": "holnap este",
  "tomorrow-night": "holnap éjjel",
  "morning": "reggel",
  "afternoon": "délután",
  "evening": "este",
  "night": "éjjel",
  "today": "ma",
  "tomorrow": "holnap",
  "sunday": "vasárnap",
  "monday": "hétfői nap",
  "tuesday": "keddi nap",
  "wednesday": "szerdai nap",
  "thursday": "csütörtöki nap",
  "friday": "pénteki nap",
  "saturday": "szombati nap",
  "next-sunday": "vasárnap", // FIXME
  "next-monday": "hétfői nap", // FIXME
  "next-tuesday": "keddi nap", // FIXME
  "next-wednesday": "szerdai nap", // FIXME
  "next-thursday": "csütörtöki nap", // FIXME
  "next-friday": "pénteki nap", // FIXME
  "next-saturday": "szombati nap", // FIXME
  "minutes": "$1 perc",
  "fahrenheit": "$1\u00B0F",
  "celsius": "$1\u00B0C",
  "inches": "$1 in.",
  "centimeters": "$1 cm.",
  "less-than": "$1 alatt",
  "and": function(a, b) {
    return join_with_shared_prefix(
      a,
      b,
      "",
    );
  },
  "through": function(a, b) {
    return join_with_shared_prefix(a, b, " között");
  },
  "with": "$1, $2",
  "range": "$1\u2013$2",
  "parenthetical": function(a, b) {
    return a + " (" + b + (a === "vegyes csapadék" ? " hó)" : ")");
  },
  "for-hour": "$1 ebben az órában",
  "starting-in": "$1 $2 múlva",
  "stopping-in": "$1 $2 múlva véget ér",
  "starting-then-stopping-later": "$1 $2 múlva, $3ig",
  "stopping-then-starting-later": "$1 $2 múlva véget ér, de $3 elteltével újra várható",
  "for-day": "$1 egész nap",
  "starting": "$1 lesz $2",
  "until": "$1 $2",
  "until-starting-again": "$1 $2, majd $3 ismét",
  "starting-continuing-until": "$1 kezdődik $2, $3 folytatódik",
  "during": "$1 $2",
  "for-week": "$1 ezen a héten",
  "over-weekend": "$1 a hétvégén",
  "temperatures-peaking": "$1 csúcshőmérséklettel $2",
  "temperatures-rising": "$1-ra emelkedő hőmérséklettel $2",
  "temperatures-valleying": "$1 minimum hőmérséklettel $2",
  "temperatures-falling": "$1-ra eső hőmérséklettel $2",
  /* Capitalize the first letter of first word. */
  "title": function(str) {
    return str.charAt(0).toUpperCase() + str.slice(1);
  },
  /* Capitalize the first word of the sentence and end with a period. */
  "sentence": function(str) {
    /* Capitalize. */
    str = str.charAt(0).toUpperCase() + str.slice(1);

    /* Add a period if there isn't already one. */
    if(str.charAt(str.length - 1) !== ".")
      str += ".";

    return str;
  },
};

File: test_cases/hu.py
# This would be one of your translations, as a Python dictionary
cases = {
    "Derült": ["title", "clear"],
    "Szitáló csapadék lehetséges": ["title", "possible-very-light-precipitation"],
    "Szitáló csapadék": ["title", "very-light-precipitation"],
    "Gyenge csapadék lehetséges": ["title", "possible-light-precipitation"],
    "Gyenge csapadék": ["title", "light-precipitation"],
    "Csapadék": ["title", "medium-precipitation"],
    "Intenzív csapadék": ["title", "heavy-precipitation"],
    "Gyenge szitálás lehetséges": ["title", "possible-very-light-rain"],
    "Gyenge szitálás": ["title", "very-light-rain"],
    "Gyenge eső lehetséges": ["title", "possible-light-rain"],
    "Gyenge eső": ["title", "light-rain"],
    "Eső": ["title", "medium-rain"],
    "Intenzív eső": ["title", "heavy-rain"],
    "Ónos szitálás lehetséges": ["title", "possible-very-light-sleet"],
    "Ónos szitálás": ["title", "very-light-sleet"],
    "Gyenge ónos eső lehetséges": ["title", "possible-light-sleet"],
    "Gyenge ónos eső": ["title", "light-sleet"],
    "Ónos eső": ["title", "medium-sleet"],
    "Havas eső": ["title", "heavy-sleet"],
    "Hószállingózás lehetséges": ["title", "possible-very-light-snow"],
    "Hószállingózás": ["title", "very-light-snow"],
    "Gyenge havazás lehetséges": ["title", "possible-light-snow"],
    "Gyenge havazás": ["title", "light-snow"],
    "Havazás": ["title", "medium-snow"],
    "Intenzív havazás": ["title", "heavy-snow"],
    "Mérsékelt szél": ["title", "medium-wind"],
    "Erős szél": ["title", "heavy-wind"],
    "Ködös idő": ["title", "fog"],
    "Erős derült": ["title", "very-light-clouds"],
    "Közepes felhősödés": ["title", "light-clouds"],
    "Erős felhősödés": ["title", "medium-clouds"],
    "Borult idő": ["title", "heavy-clouds"],
    "Száraz idő és enyhe szél": ["title", ["and", "low-humidity", "light-wind"]],
    "Gyenge szitálás és erős szél": ["title", ["and", "very-light-rain", "heavy-wind"]],
    "Párás idő és közepes felhősödés": [
        "title",
        ["and", "high-humidity", "light-clouds"],
    ],
    "Derült ebben az órában.": ["sentence", ["for-hour", "clear"]],
    "Hószállingózás 35 perc múlva.": [
        "sentence",
        ["starting-in", "very-light-snow", ["minutes", 35]],
    ],
    "Gyenge eső 15 perc múlva véget ér.": [
        "sentence",
        ["stopping-in", "light-rain", ["minutes", 15]],
    ],
    "Havas eső 20 perc múlva, 30 percig.": [
        "sentence",
        [
            "starting-then-stopping-later",
            "heavy-sleet",
            ["minutes", 20],
            ["minutes", 30],
        ],
    ],
    "Eső 25 perc múlva véget ér, de 8 perc elteltével újra várható.": [
        "sentence",
        [
            "stopping-then-starting-later",
            "medium-rain",
            ["minutes", 25],
            ["minutes", 8],
        ],
    ],
    "Erős felhősödés egész nap.": ["sentence", ["for-day", "medium-clouds"]],
    "Ónos szitálás lesz reggel.": [
        "sentence",
        ["starting", "very-light-sleet", "morning"],
    ],
    "Mérsékelt szél ma éjjel.": ["sentence", ["until", "medium-wind", "today-night"]],
    "Intenzív csapadék délután.": [
        "sentence",
        ["until", "heavy-precipitation", "afternoon"],
    ],
    "Enyhe szél délután.": ["sentence", ["during", "light-wind", "afternoon"]],
    "Havazás ma késő este és holnap reggel.": [
        "sentence",
        ["during", "medium-snow", ["and", "later-today-evening", "tomorrow-morning"]],
    ],
    "Intenzív eső ma késő délelőtt, majd ma este ismét.": [
        "sentence",
        ["until-starting-again", "heavy-rain", "later-today-morning", "today-evening"],
    ],
    "Borult idő kezdődik este, éjjel folytatódik.": [
        "sentence",
        ["starting-continuing-until", "heavy-clouds", "evening", "night"],
    ],
    "Gyenge ónos eső ma késő délután és ködös idő holnap reggel.": [
        "sentence",
        [
            "and",
            ["during", "light-sleet", "later-today-afternoon"],
            ["during", "fog", "tomorrow-morning"],
        ],
    ],
    "Erős szél kezdődik ma reggel, ma délután folytatódik és ónos eső holnap reggel.": [
        "sentence",
        [
            "and",
            [
                "starting-continuing-until",
                "heavy-wind",
                "today-morning",
                "today-afternoon",
            ],
            ["during", "medium-sleet", "tomorrow-morning"],
        ],
    ],
    "Borult idő lesz ma késő éjjel és intenzív havazás holnap délután.": [
        "sentence",
        [
            "and",
            ["starting", "heavy-clouds", "later-today-night"],
            ["during", "heavy-snow", "tomorrow-afternoon"],
        ],
    ],
    "Száraz idő ma éjjel és gyenge csapadék kezdődik holnap este, holnap éjjel folytatódik.": [
        "sentence",
        [
            "and",
            ["during", "low-humidity", "today-night"],
            [
                "starting-continuing-until",
                "light-precipitation",
                "tomorrow-evening",
                "tomorrow-night",
            ],
        ],
    ],
    "Havazás (5 in.) éjjel.": [
        "sentence",
        ["during", ["parenthetical", "medium-snow", ["inches", 5]], "night"],
    ],
    "Gyenge havazás (2 cm.) ma késő délelőtt.": [
        "sentence",
        [
            "during",
            ["parenthetical", "light-snow", ["centimeters", 2]],
            "later-today-morning",
        ],
    ],
    "Intenzív havazás (8\u201312 in.) egész nap.": [
        "sentence",
        ["for-day", ["parenthetical", "heavy-snow", ["inches", ["range", 8, 12]]]],
    ],
    "Havazás (1 cm. alatt) délután.": [
        "sentence",
        [
            "during",
            ["parenthetical", "medium-snow", ["less-than", ["centimeters", 1]]],
            "afternoon",
        ],
    ],
    "Csapadékmentes idő ezen a héten, 85\u00b0F csúcshőmérséklettel holnap.": [
        "sentence",
        [
            "with",
            ["for-week", "no-precipitation"],
            ["temperatures-peaking", ["fahrenheit", 85], "tomorrow"],
        ],
    ],
    "Vegyes csapadék a hétvégén, 32\u00b0C-ra emelkedő hőmérséklettel csütörtöki nap.": [
        "sentence",
        [
            "with",
            ["over-weekend", "mixed-precipitation"],
            ["temperatures-rising", ["celsius", 32], "thursday"],
        ],
    ],
    "Gyenge szitálás hétfői nap, 15\u00b0F minimum hőmérséklettel pénteki nap.": [
        "sentence",
        [
            "with",
            ["during", "very-light-rain", "monday"],
            ["temperatures-valleying", ["fahrenheit", 15], "friday"],
        ],
    ],
    "Gyenge havazás keddi nap és szerdai nap, 0\u00b0C-ra eső hőmérséklettel vasárnap.": [
        "sentence",
        [
            "with",
            ["during", "light-snow", ["and", "tuesday", "wednesday"]],
            ["temperatures-falling", ["celsius", 0], "sunday"],
        ],
    ],
    "Csapadék ma és szombati nap között, 100\u00b0F csúcshőmérséklettel hétfői nap.": [
        "sentence",
        [
            "with",
            ["during", "medium-precipitation", ["through", "today", "saturday"]],
            ["temperatures-peaking", ["fahrenheit", 100], "monday"],
        ],
    ],
    "Vegyes csapadék (1\u20133 in. hó) egész nap.": [
        "sentence",
        [
            "for-day",
            ["parenthetical", "mixed-precipitation", ["inches", ["range", 1, 3]]],
        ],
    ],
    "Intenzív havazás (1\u20133 in.)": [
        "title",
        ["parenthetical", "heavy-snow", ["inches", ["range", 1, 3]]],
    ],
    "Intenzív havazás (3\u20135 cm.)": [
        "title",
        ["parenthetical", "heavy-snow", ["centimeters", ["range", 3, 5]]],
    ],
}
